### DISCUSSION TITLE
Can't seem to get runtime config values

### DISCUSSION THREAD

**Discussion by HampusLilja** [nuxt/nuxt#17570] [Questions] [answered]
Hi,

I'm trying to access an environment variable in a middleware but I might be doing it wrong because I can't seem to get the value that is defined in the .env file in any file/component.

In this stackblitz I can see that the value is set on the process.env but the useRuntimeConfig has the default value from nuxt.config. 
Why isn't the runtimeConfig value the same as the process.env value?

https://stackblitz.com/edit/github-hw4l2j-hnlr42?file=middleware/featureToggle.global.ts

**Middleware:**
`export default defineNuxtRouteMiddleware((_, __) => {
  const config = useRuntimeConfig();
  console.log('middle-conf', config.public.FEATURE_ENABLED);
  console.log('middle-env', process.env?.FEATURE_ENABLED);
});`

**nuxt.config**
`import { defineNuxtConfig } from 'nuxt';
// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  runtimeConfig: {
    public: {
      FEATURE_ENABLED: 'false',
    },
  },
});`

**.env**
`FEATURE_ENABLED=true`

Help is much appreciated :)

**Comment by HampusLilja** [comment]
As I understand it, the values in nuxt config are just default values and will be replaced by the values in the .env when the app is running?

**Comment by danielroe** [comment, marked as answer]
Check out https://v3.nuxtjs.org/guide/features/runtime-config. Nuxt doesn't read `.env` in production - you will need to explicitly set them in your environment. And the syntax for overriding would be `NUXT_PUBLIC_FEATURE_ENABLED=true node .output/server/index.mjs`.

**Reply by HampusLilja** [reply]
Aah ofc, I'm just mixing up the different environments I have locally.
Thanks for a quick and clear answer!

**Reply by ghost** [reply]
It would be nice to specify this in the documentation. There is no text that says that the .env file will be ignored in production

**Reply by ivaneliasoo** [reply]
thanks, this save my day!
You're right in the nuxt.config.ts docs specifically says **PUBLIC_BASE_URL** not  **NUXT_PUBLIC_BASE_URL**  but it's ok here https://v3.nuxtjs.org/guide/features/runtime-config

**Reply by TheAlexLichter** [reply]
> It would be nice to specify this in the documentation. There is no text that says that the .env file will be ignored in production

Should be part of it now 👍
